wire.md

```diff
@@ -0,0 +1,1008 @@
+the behalf of a requester, in order to reduce redundant retransmission of post
+or hash data (Post Response and Hash Response, respectively).
+
+For example, consider a client `A` which sends a request to `B` with `ttl = 1`.
+`B` then forwards that request to their peers `C` and `D`. If `B` responds to `A`
+with a set of N hashes or posts `B` could track what they sent. Additionally, in the
+case that `C` or `D`'s responses -- routed through `B` -- contain any
+duplicates that `B` knows were already sent back to `A`, `B` could choose to
+edit these response messages, to omit the duplicates from being needlessly
+retransmitted onwards back to `A`.
+
+Regarding the above section (5.2.3 Limits), hashes that are deduplicated by an
+intermediary peer, and thus not transmitted back to the requester, do not count
+against the `limit`.
+
+An intermediary peer MAY also elect to perform deduplication in the other
+direction on Post Requests, by first checking their local database to see which
+hashes they already have posts for, and editing the original Post Request
+message to no longer request those posts, before forwarding that message on to
+further peers.
+
+### 5.3 Users
+
+#### 5.3.1 Names
+- A valid user name MUST be a UTF-8 string.
+- A valid user name MUST between 1 and 32 codepoints.
+
+#### 5.3.2 State
+A user is described by the following:
+
+1. their public key
+2. the key/value pairs of the latest known `post/info` post made by that user.
+
+As of present, the only supported key is `name`, which defines a user's display
+name.
+
+`post/info` posts older than the latest for a user are considered obsolete.
+
+### 5.4 Channels
+A channel is a named collection consisting of the following:
+
+1. chat messages (`post/text`)
+2. user joins and leaves (`post/join` or `post/leave`).
+3. a topic string (`post/topic`)
+
+A user writing a chat message or join to a channel implies that that named
+channel has now been created, if it hasn't already, and thus MUST be returned
+in Channel List Requests.
+
+#### 5.4.1 Names
+- A valid channel name MUST be a UTF-8 string.
+- A valid channel name MUST between 1 and 64 codepoints.
+
+#### 5.4.2 Topics
+- A valid channel topic MUST be a UTF-8 string.
+- A valid channel topic MUST be between 0 and 512 codepoints.
+- If there is no known topic set for a channel, it MUST be considered the empty string (`""`).
+- A channel topic string set by a user to the empty string MUST be considered as there being no topic currently set.
+
+#### 5.4.3 User Membership
+A user makes a post "to a channel" if they have set the `channel` field on said
+post to the name of that channel.
+
+A user is a member of a channel at a particular point in time if, from a
+client's perspective, that user has issued a `post/join`, `post/text`, or
+`post/topic` to that channel and has not issued a `post/leave` since.
+
+If a user's latest known post to a channel is a `post/leave`, they are not a
+member of that channel.
+
```

Comment:
Then they are a member, as per the preceding paragraph:

> A user is a member of a channel at a particular point in time if, from a host's perspective, that user has issued a `post/join`, `post/text`, or `post/topic` to that channel and has not issued a `post/leave` since.

Reply:
I've revised this to more simply state, "Otherwise, they are not considered a member."